go/enclave/rollupchain/rollup_chain.go

```diff
@@ -50,10 +52,11 @@ type RollupChain struct {
 	rpcEncryptionManager  rpcencryptionmanager.RPCEncryptionManager
 	mempool               mempool.Manager
 
+	privateKey           *ecdsa.PrivateKey
```

Comment:
Perhaps worth a comment on this field to make it clear that it's the enclave's private key that's getting used for signing the headers